plat/mediatek/common/mtk_sip_svc.h

```diff
@@ -33,20 +33,58 @@
 #include <stdint.h>
 
 /* SMC function IDs for SiP Service queries */
-#define SIP_SVC_CALL_COUNT		0x8200ff00
-#define SIP_SVC_UID			0x8200ff01
-/*					0x8200ff02 is reserved */
-#define SIP_SVC_VERSION			0x8200ff03
+#define SIP_SVC_CALL_COUNT    0x8200ff00
+#define SIP_SVC_UID           0x8200ff01
+/* 0x8200ff02 is reserved */
+#define SIP_SVC_VERSION       0x8200ff03
 
 /* Mediatek SiP Service Calls version numbers */
-#define MTK_SIP_SVC_VERSION_MAJOR	0x0
-#define MTK_SIP_SVC_VERSION_MINOR	0x1
+#define MTK_SIP_SVC_VERSION_MAJOR 0x0
+#define MTK_SIP_SVC_VERSION_MINOR 0x1
 
 /* Number of Mediatek SiP Calls implemented */
-#define MTK_SIP_NUM_CALLS		1
+#define MTK_SIP_NUM_CALLS   1
 
 /* Mediatek SiP Service Calls function IDs */
-#define MTK_SIP_SET_AUTHORIZED_SECURE_REG	0x82000001
+#define MTK_SIP_SET_AUTHORIZED_SECURE_REG 0x82000001
+
+/* For MTK SMC from Secure OS */
+/* 0x82000000 - 0x820000FF & 0xC2000000 - 0xC20000FF */
+#define MTK_SIP_TBASE_HWUID_AARCH32         0x82000000
+#define MTK_SIP_LK_WDT_AARCH32              0x82000106
+#define MTK_SIP_LK_WDT_AARCH64              0xC2000106
+/* astone added for debug purpose */
+#define MTK_SIP_KERNEL_MSG_AARCH32          0x820002ff
+#define MTK_SIP_KERNEL_MSG_AARCH64          0xC20002ff
+#define MTK_SIP_KERNEL_WDT_AARCH32          0x82000204
+#define MTK_SIP_KERNEL_WDT_AARCH64          0xC2000204
+#define MTK_SIP_KERNEL_BOOT_AARCH32         0x82000200
+#define MTK_SIP_KERNEL_BOOT_AARCH64         0xC2000200
+/*
+ *Please follow the naming rule
+ *#define SIP_SVC_E_XXX error_num (should be negative value)
+ */
+#define SIP_SVC_E_SUCCESS           0
+#define SIP_SVC_E_NOT_SUPPORTED     -1
+#define SIP_SVC_E_INVALID_PARAMS    -2
+#define SIP_SVC_E_INVALID_Range     -3
+#define SIP_SVC_E_PERMISSION_DENY   -4
+#define SIP_SVC_E_LOCK_FAIL         -5
+
+struct kernel_info {
+	uint64_t pc;
+	uint64_t r0;
+	uint64_t r1;
+	uint64_t r2;
+	uint64_t k32_64;
+};
+
+typedef struct mtk_bl_param {
```

Comment:
This structure is not related to SiP services. It should be moved to a common file.